### ISSUE TITLE
[Feature request]: Incrementally increasing complexity of subsequent postponements of the same long break

### ISSUE THREAD

**Issue by AKagitin** [hovancik/stretchly#1492]
### Version

- [ ] I'm using version 1.16

### Existing requests

- [X] I've checked Existing requests

### Advanced Preferences

- [X] I've checked Advanced Preferences

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### Problem description

  I'm always frustrated I understand that have postponed a break. But sometimes it is necessary, so blocking of this possibility is not a good solution. 

### Possible solution

I think that the best way is to make more complex to postpone the same break in 2nd, 3rd, ..., time in the following fashion:
User must to enter a number how many times he/she already have postponed this break. I think this variant will be good as it will encourage the user to remember his actions and so to better control his/her behavior. If he/she fails then the break will last as planned accordingly to settings.

Thank you for the good app!
With best regards, Anatoly

### Alternative solutions

User must to click the "Skip..." button more than one time (incrementally on each user's decision to postpone the same break).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hovancik** [comment]
Not exactly what you ask, but you can play a bit with how postpone works currently: https://github.com/hovancik/stretchly?tab=readme-ov-file#editing-postpone-functionality-

**Comment by GeeWee** [comment]
LeechBlock NG has a feature where you e.g. have to input a random string to turn it off for a bit - would you be interested in a contribution that did something similar, where you had to input a random string or solve a simple math puzzle?

**Comment by hovancik** [comment]
@GeeWee could you please explain a bit more what you mean?

**Comment by GeeWee** [comment]
Yes of course - sorry for perhaps being a little brief before. In Leechblock NG, you can turn off a feature that essentially gets you this screen if you want to turn it off

![Image](https://github.com/user-attachments/assets/7dc67867-ecac-41b2-b56b-379bcfa54702)

(The length is customizable)

I'm proposing we could add that as an option for skipping breaks, so that if you want to skip a break you have to either solve a simple math question or type a small string, to make it harder (but still possible) to skip the break.

**Comment by hovancik** [comment]
Gotcha :) How would you see it interacting with current Strict mode?

**Comment by GeeWee** [comment]
I think I foresee it as an in-between? I.e. it's less restrictive than strict mode, so I guess I'd expect it to not work if strict mode is turned on, but that's just a gut feel :)

**Comment by AKagitin** [comment]
> to skip a break you have to either solve a simple math question or type a small string

I think this is not the best solution. It will be bad system behaviour at the moments when user is having urgent discussion/work in contact with colleagues and/or bosses, for example. At such moments imho system is welcome to remind about breaks even more frequently but shouldn't interrupt current conversation. I think that the best and simplest solution is to incrementally reduce postponement interval down to 1 minute.  

To manage this mode in combination with Strict mode you could add one new checkbox to setup so it looks like this:
Strict mode:
Strict mode prevents you from skipping either Mini breaks or Long breaks and is designed to help discipline.

[v] Enable Strict mode for Mini breaks
[v] Enable Strict mode for Long breaks
[v] Enable automatic reducing postponement interval by 1 minute at every sequential postponement. Only for Long breaks. If Strict mode is enabled it will have effect after you have used all postponement times for the current break.

**Comment by hovancik** [comment]
@AKagitin you mean instead of 5 minites, it would go to 4, then 3 then 2 etc?  @GeeWee what do you think? 

In general, I think this: for different people different things seems to work. Therefore I am very carefull with adding anything new in this area because it gets complicated very quickly.

**Comment by GeeWee** [comment]
Essentially I just want something that makes it harder to keep clicking "postpone" via muscle memory, but without forcing me to make the break unskippable, as that's e.g. a problem if I'm in a meeting. I'm in favour of whatever implementation that moves in that direction :)
